### ISSUE TITLE
Cache Texture to HDD Crash

### ISSUE THREAD

**Issue by shinra358** [gonetz/GLideN64#1290]
P570WM
I7-4960x
980M
32gb 1600hz ram

https://www.youtube.com/watch?v=YUJbZbHRRGQ&t=34s

When a texture pack is loaded initially, and cache textures to HDD is on, and you end the emulation, the textures will start to cache to the hdd, but when it gets near the end, the emulator will crash @8:45.

@2:21: Reloading the emu and game and then quitting emulation will make the cache textures to disk words show, but they wont be moving. But if you click around the emulator, pieces of the emu will reveal itself from underneath the black screen. During this time, there will be no close or minimize buttons on the emu.

**Comment by gonetz** [comment]
> When a texture pack is loaded initially ... the emulator will crash 

Do you have that problem with **any** texture pack? I just tried to load few packs I have, no issues with load and saving to disk.

**Comment by shinra358** [comment]
I only have the zelda pack.

**Comment by gonetz** [comment]
the zelda pack?! I know at least 3 Zelda packs. Which one is "the pack"?

**Comment by shinra358** [comment]
Mainly the community one. But i have a tiny bit of various others mixed in
to improve certain textures.

**Comment by gonetz** [comment]
Could you upload your pack somewhere? I tested community pack few months ago, and met no problems.

**Comment by shinra358** [comment]
When your pack loads and games starts, did you press f12? If not do that.
Uploading pack now.

Forgot I had a majora's mask pack by bluedart and it does the same thing. I really don't think it's a texture pack prob but we'll see.

**Comment by gonetz** [comment]
I can't load that 700mb from Mega: it says limit is exceeded.

**Comment by gonetz** [comment]
I found in my archives 730mb file THE LEGEND OF ZELDA V7.7z I guess it is some version of community pack. It failed to load with error: failed to allocate memory. The pack needs more than 1GB of memory. As I remember, 32bit windows application can't get more than 2GB from OS. The packs just don't feet the memory. If you really need to use it, you need 64 bit version of emulator and plugin. 64 bit version of GLideN64 is currently available only for Linux.

**Comment by shinra358** [comment]
I got past that by using the 4gb patcher on the project64 executable.

**Comment by gonetz** [comment]
I was lucky to load the pack once, and it plugin really crashed at the end of saving process.
It hardly will be fixed soon.
BTW, there are other packs for Zelda, they work without issues.

**Comment by shinra358** [comment]
So you used the 4gb patcher? And what about the other packs like the
majoras mask one that does the same thing? And how can i test properly for
texture corruption if this isnt fixed? ive had this pack for the longest
and its the most popular pack. I picked the pack based on realism and no
other pack is like that. Aint no way a person can make a good pack under
2gb :P

On Dec 7, 2016 12:23 PM, "Sergey Lipskiy" <notifications@github.com> wrote:

I was lucky to load the pack once, and it plugin really crashed at the end
of saving process.
It hardly will be fixed soon.
BTW, there are other packs for Zelda, they work without issues.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/gonetz/GLideN64/issues/1290#issuecomment-265512230>,
or mute
the thread

**Comment by LuismaSP89** [comment]
I have the same pack as you "Original v7.7 of the community" and works ok, it doesn´t crash. (Using too the 4gb patch, of course)

Tested too with the pack you provide ---> Works ok.

The only problem is that GlideN64 can´t load textures with transparency and fails in certain textures (Images attached)

![gliden64_the_legend_of_zelda_000](https://cloud.githubusercontent.com/assets/12205048/21237068/b74ab4dc-c2fd-11e6-81d4-fc0c8bbe7faa.jpg)
![gliden64_the_legend_of_zelda_001](https://cloud.githubusercontent.com/assets/12205048/21237071/b74b15b2-c2fd-11e6-8544-c1f28dd491fa.jpg)
![gliden64_the_legend_of_zelda_003](https://cloud.githubusercontent.com/assets/12205048/21237072/b74be816-c2fd-11e6-9852-1fb598d709fd.jpg)
![gliden64_the_legend_of_zelda_004](https://cloud.githubusercontent.com/assets/12205048/21237069/b74adb4c-c2fd-11e6-9295-a09f3a812999.jpg)
![gliden64_the_legend_of_zelda_005](https://cloud.githubusercontent.com/assets/12205048/21237070/b74b25c0-c2fd-11e6-83dd-2f92a5c48dd5.jpg)


Btw, can you provide your glideN64 settings to see where´s the problem?

**Comment by LuismaSP89** [comment]
Oops, sorry, i see now your video attached.

Try this:

vram usage limit: 999
enhanced texture cache size: 999

Else... can be the reshade?

**Comment by Dogstalotl** [comment]
I'm having the same issue but with another game with other settings and different parts.
(Specs:
GA-Z97X-Gaming 7
Geforce GTX 960
Intel Core i5-4690K
16 GB DDR3 RAM)

I basically am trying to load Star Fox 64 with a texture pack combination of [razius's Star Fox 64 HD Texture Pack](http://www.emutalk.net/threads/53915-Starfox-64-(U)-(fonts-particles-and-HUD)?p=445028&viewfull=1#post445028) and [Lylat Reloaded](http://n64-modding.darkbb.com/f3-download)

Here's a download if needed:
[http://www.mediafire.com/file/g54bnz72bb66jlb/STARFOX64.zip](http://www.mediafire.com/file/g54bnz72bb66jlb/STARFOX64.zip)

Also, here's some screenshots of my configurations: 
[http://imgur.com/a/Ng5mc](http://imgur.com/a/Ng5mc)

Can somebody help me?

**Comment by Dogstalotl** [comment]
On further inspection, these settings cause all other games not to work.

I changed the Enhanced Texture Size down to 511MB, the other games loaded but the error persists in StarFox 64. Then, while the size was lowered, I turned off Copy auxiliary control buffers to N64 memory and the other games still worked but the bug for StarFox lasted longer before giving up with the error.

I then reset it all back to normal and disabled the texture cache and other games worked but I still received the error in StarFox 64. So I set it back again.

I then tried disabling Anti-aliasing and Anisotropic filtering but it crashed on both with the same result... I am now starting to think that this problem is in the Enhanced Texture Size/the Texture Cache.

So I turned them back on and tried disabling "Read color buffer in 4kb chunks". They all crashed.

I set it back once again and in a last ditch effort. I disabled the frame buffer... all the games didn't work.

I tried everything I could think of, and now I am lead to believe that this problem is in the Enhanced Texture Size/the Texture Cache. This may need a patch. (No rhyme intended)
